admin/sql/eaa/CreateMQTriggers.sql

```diff
@@ -0,0 +1,173 @@
+BEGIN;
+
+SET search_path = 'event_art_archive';
+
+CREATE OR REPLACE FUNCTION reindex_event() RETURNS trigger AS $$
+    DECLARE
+        event_mbid UUID;
+    BEGIN
+        SELECT gid INTO event_mbid
+        FROM musicbrainz.event e
+        JOIN event_art_archive.event_art ea ON e.id = ea.event
+        WHERE e.id = NEW.id;
+
+        IF FOUND THEN
+            PERFORM amqp.publish(1, 'event-art-archive', 'index', event_mbid::text);
+        END IF;
+        RETURN NULL;
+    END;
+$$ LANGUAGE 'plpgsql';
+
+CREATE TRIGGER eaa_reindex AFTER UPDATE OR INSERT
+ON musicbrainz.event FOR EACH ROW
+EXECUTE PROCEDURE reindex_event();
+
+CREATE OR REPLACE FUNCTION reindex_artist() RETURNS trigger AS $$
+    BEGIN
+        -- Short circuit if the name hasn't changed
+        IF NEW.name = OLD.name AND NEW.sort_name = OLD.sort_name THEN
+            RETURN NULL;
+        END IF;
+
+        PERFORM amqp.publish(1, 'event-art-archive', 'index', e.gid::text)
+        FROM musicbrainz.event e
+        JOIN event_art_archive.event_art ea ON e.id = ea.event
+        JOIN musicbrainz.l_artist_event lae ON e.id = lae.entity1
+        WHERE lae.entity0 = NEW.id;
+
+        RETURN NULL;
+    END;
+$$ LANGUAGE 'plpgsql';
+
+CREATE TRIGGER eaa_reindex AFTER UPDATE
+ON musicbrainz.artist FOR EACH ROW
+EXECUTE PROCEDURE reindex_artist();
+
+CREATE OR REPLACE FUNCTION reindex_l_artist_event() RETURNS trigger AS $$
```

Comment:
Forgot to explain these. In the IA slack, for the *_mb_metadata.xml files we settled on sending the place XML with artist-rels and place-rels included (since at a minimum it'd probably be useful for the IA to show "who" and "where" on their details pages). So this will have it reindex whenever those relationships change. It doesn't reindex if, say, an attribute on one of the relationships changes. We *could* add that in the future, if they actually use attributes somehow, but it seems overkill for now.

Reply:
Did you consider joining event-rels (9K) and series-rels (18K) which are mostly about “parts”?

Also, it would be nice to allow event art for event series (tour, festival, run).

Reply:
> Did you consider joining event-rels (9K) and series-rels (18K) which are mostly about “parts”?

Not really. But I think we should wait and see what the IA does with their details pages, 'cause we don't want to trigger on changes for data they don't use. We could add new triggers/functions (or even modify the existing ones) after the schema change, since these have to be created manually anyway, and aren't part of the core schema - nobody will have these in their DB unless they're hacking on CAA stuff.

> Also, it would be nice to allow event art for event series (tour, festival, run).

Yeah, probably for a future schema change. :P Or we can have an event art type for it and put it on the first event in the series for now.

Reply:
> Also, it would be nice to allow event art for event series (tour, festival, run).

Keep in mind that for festivals at least, each year's festival is its own main event (so a festival poster can be uploaded there). Agreed that a tour poster would be nice to have at the series level, but it doesn't seem super important just yet :)

Reply:
By the way, this feature is also missing for CAA and release (group) series.

Reply:
What would be the use of that? The series logo or something? That's already more like https://tickets.metabrainz.org/browse/MBS-4683